pyuvsim/simsetup.py

```diff
@@ -988,31 +988,54 @@ def initialize_catalog_from_params(
         return sky
 
 
+def _check_uvbeam_file(beam_file):
+    if not os.path.exists(beam_file):
+        testdata_beam_file = os.path.join(SIM_DATA_PATH, beam_file)
+        if os.path.exists(testdata_beam_file):
+            beam_file = testdata_beam_file
+        else:
+            raise ValueError(f"Unrecognized beam file or model: {beam_file}")
+    return beam_file
+
+
 def _construct_beam_list(beam_ids, telconfig, freq_range=None, force_check=False):
     """Construct BeamList."""
     beam_list = []
+    uvb_read_kwargs = {}
     for beamID in beam_ids:
         beam_model = telconfig['beam_paths'][beamID]
 
         if not isinstance(beam_model, (str, dict)):
             raise ValueError('Beam model is not properly specified in telescope config file.')
 
-        # first check to see if the beam_model is a string giving a file location
-        if (isinstance(beam_model, str)
-                and (os.path.exists(beam_model)
-                     or os.path.exists(os.path.join(SIM_DATA_PATH, beam_model)))):
-            if os.path.exists(beam_model):
-                beam_list.append(beam_model)
-            else:
-                beam_list.append(os.path.join(SIM_DATA_PATH, beam_model))
-        # Failing that, try to parse the beam string as an analytic beam.
+        if isinstance(beam_model, str) and beam_model not in AnalyticBeam.supported_types:
+            # this is the path to a UVBeam readable file
+            beam_file = _check_uvbeam_file(beam_model)
+
+            beam_list.append(beam_file)
+        elif "filename" in beam_model:
```

Comment:
In the lines above this if/elif/else block we check that beam_model is either a string or a dict. The first `if` checks if it's a string, so `elif` this can safely assume it's a dict.